### PR TITLE
Pass MO meta data to RT info

### PR DESCRIPTION
**Pull request by popovaan**
Description: 
Added passing of MO meta data to RT info.

Ticket: 88606

Code:
* [x]  Comments
* [x]  Code style (PEP8)
* [x]  Transformation generates reshape-able IR
* [x]  Transformation preserves original framework node names
* [x]  Transformation preserves tensor names

Validation:
* [x]  Unit tests
* [x]  Framework operation tests
* [x]  Transformation tests
* [x]  Model Optimizer IR Reader check

Documentation:
* [x]  Supported frameworks operations list
* [x]  Guide on how to convert the **public** model
* [x]  User guide update

### CODE REVIEW

File: tools/mo/openvino/tools/mo/convert_impl.py [hunk 1]
```diff
@@ -578,12 +581,20 @@ def _convert(**args):
         init_logger('ERROR', False)
 
         argv.feManager = FrontEndManager()
-        ngraph_function = driver(argv)
+        ov_model, legacy_path = driver(argv)
+
+        # add MO meta data to model
+        ov_model.set_rt_info(get_version(), "MO_version")
+        ov_model.set_rt_info(get_rt_version(), "Runtime_version")
+        ov_model.set_rt_info(str(legacy_path), "legacy_path")
```
Comment (ilyachur) [inline tools/mo/openvino/tools/mo/convert_impl.py:589]: Do we need this variable?

File: tools/mo/openvino/tools/mo/convert_impl.py [hunk 2]
```diff
@@ -578,12 +581,20 @@ def _convert(**args):
...
+        meta_info = get_meta_info(argv)
+        for key, value in meta_info.items():
+            ov_model.set_rt_info(str(value), ["conversion_parameters", str(key)])
```
Comment (ilyachur) [inline tools/mo/openvino/tools/mo/convert_impl.py:592]: @slyalin also do we need all keys from meta info?

File: tools/mo/unit_tests/mo/convert/meta_data_test.py [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+            onnx_net = helper.make_model(graph_def, producer_name='test_model')
+            return onnx_net
+
+        def ref_meta_data():
+            return {
+                'MO_version': get_version(),
+                'Runtime_version': get_rt_version(),
+                'legacy_path': "False",
+                'conversion_parameters': {
+                    'caffe_parser_path': Path("DIR"),
```
Comment (ilyachur) [inline tools/mo/unit_tests/mo/convert/meta_data_test.py:67]: Do we need it?

File: tools/mo/unit_tests/mo/convert/meta_data_test.py [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+                    'reverse_input_channels': "False",
+                    'save_params_from_nd': "False",
+                    'scale_values': "()",
+                    'silent': "True",
+                    'source_layout': "()",
+                    'static_shape': "False",
+                    'stream_output': "False",
+                    'target_layout': "()",
+                    'transform': "",
+                    'unset': "['input_shape', 'scale', 'input', 'output', 'disable_fusing', 'finegrain_fusing', "
```
Comment (ilyachur) [inline tools/mo/unit_tests/mo/convert/meta_data_test.py:105]: ?

File: tools/mo/unit_tests/mo/convert/meta_data_test.py [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+                             "'batch', 'transformations_config', 'input_checkpoint', 'input_meta_graph', "
+                             "'saved_model_dir', 'saved_model_tags', 'tensorflow_custom_operations_config_update', "
+                             "'tensorflow_use_custom_operations_config', "
+                             "'tensorflow_object_detection_api_pipeline_config', 'tensorboard_logdir', "
+                             "'tensorflow_custom_layer_libraries', 'input_proto', 'mean_file', 'mean_file_offsets', "
+                             "'input_symbol', 'nd_prefix_name', 'pretrained_model_name', 'counts', "
+                             "'placeholder_shapes']",
+                    'use_legacy_frontend': "False",
```
Comment (ilyachur) [inline tools/mo/unit_tests/mo/convert/meta_data_test.py:113]: Do we need all this information in new representation?

**Review by ilyachur — COMMENTED** [review]
Do we need to change IRReader or IREngine?

File: tools/mo/openvino/tools/mo/convert_impl.py (hunk 1 above)
Comment (popovaan) [reply tools/mo/openvino/tools/mo/convert_impl.py:589]: We need it for debugging, it shows whether IR was generated using frontend or legacy path.

**Comment by popovaan** [comment]
> Do we need to change IRReader or IREngine?

Yes, we need. Updated MO IR reader for reading of net rt_info.

File: tools/mo/openvino/tools/mo/convert_impl.py (hunk 2 above)
Comment (popovaan) [reply tools/mo/openvino/tools/mo/convert_impl.py:597]: Discussed it offline with @slyalin, we don't need keys that have default values or unset.

File: tools/mo/unit_tests/mo/convert/meta_data_test.py (hunk 4 above)
Comment (popovaan) [reply tools/mo/unit_tests/mo/convert/meta_data_test.py]: We don't need it, I removed it.

File: tools/mo/unit_tests/mo/convert/meta_data_test.py (hunk 5 above)
Comment (popovaan) [reply tools/mo/unit_tests/mo/convert/meta_data_test.py]: This value will be in meta data only if it is set to non-default value by user.

File: tools/mo/unit_tests/mo/convert/meta_data_test.py (hunk 3 above)
Comment (popovaan) [reply tools/mo/unit_tests/mo/convert/meta_data_test.py]: This value will be in meta data only if it is set to non-default value by user.

**Comment by ilyachur** [comment]
@rkazants @slyalin Please take a look, we need to have it in 2022.3 releae

File: tools/mo/openvino/tools/mo/utils/cli_parser.py [hunk 6]
```diff
@@ -2225,6 +2225,13 @@ def depersonalize(value: str, key: str):
     dir_keys = [
         'output_dir', 'extensions', 'saved_model_dir', 'tensorboard_logdir', 'caffe_parser_path'
     ]
+
+    if isinstance(value, list):
```
Comment (mvafin) [inline tools/mo/openvino/tools/mo/utils/cli_parser.py:2229]: If `value` can be `list` you need to change function signature, currently value is stated as `str`.

**Review by mvafin — APPROVED** [review]
Looks good to me, small comment

**Pull request merged**